src/resolvers/question.ts

```diff
@@ -0,0 +1,181 @@
+import { Resolvers } from "../types";
+import { MyContext } from "../context";
+import { Question } from "../models/Question";
+import { Section } from "../models/Section";
+import { hasPermissionOnQuestion } from "../services/questionService";
+import { ForbiddenError, NotFoundError, BadUserInput } from "../utils/graphQLErrors";
+
+
+export const resolvers: Resolvers = {
+  Query: {
+    questions: async (_, { sectionId }, context: MyContext): Promise<Question[]> => {
+      const section = await Section.findById('questions resolver', context, sectionId);
+      if (await hasPermissionOnQuestion(context, section.templateId)) {
+        return await Question.findBySectionId('questions resolver', context, sectionId);
+      }
+      throw ForbiddenError();
+    },
+    question: async (_, { questionId }, context: MyContext): Promise<Question> => {
+
+      // Find question with questionId
+      const question = await Question.findById('section resolver', context, questionId);
+
+      if (await hasPermissionOnQuestion(context, question.templateId)) {
+        return question;
+      }
+      throw ForbiddenError();
+    }
+  },
+  Mutation: {
+    addQuestion: async (_, { input: {
+      templateId,
```

Comment:
@briri - possibly, although it's helpful to me to have all the properties immediately viewable while looking at the mutation, instead of having it abstracted away to another location.

Reply:
👍🏻 I'm ok with that as well and agree that sometimes you can have too much abstraction. I'll follow the same pattern going forward